tests/test_core.py

```diff
@@ -65,6 +65,49 @@ def test_init_df_with_extra_col(test_pd_df):
                                   tdf, check_like=True)
 
 
+@pytest.mark.xfail(reason=(
+    "pandas datetime is limited to the time period of ~1677-2262, see "
+    "https://stackoverflow.com/a/37226672"
+))
+def test_init_df_long_timespan(test_pd_df):
```

Comment:
a little pedantic, but could you make three tests named

- `test_init_datetime`
- `test_init_datetime_subclass`
- `test_init_datetime_long_timespan`

And possibly a short comment noting why the subclass test passes

Reply:
nice suggestion, done